__tests__/erebos-api-bzz-node.js

```diff
@@ -97,4 +101,21 @@ describe('Bzz', () => {
     const buffer = await bzz.downloadRawBuffer(manifestHash)
     expect(buffer.toString('utf8')).toBe(uploadContent)
   })
+
+  it('uploadDirectory() uploads the contents and returns the hash of the manifest', async () => {
+    const dir = {
+      "foo.txt": {data: "this is foo.txt"},
+      "bar.txt": {data: "this is bar.txt"}
+    }
+    const dirHash = await bzz.uploadDirectory(dir)
+    const manifest = await bzz.downloadRawText(dirHash)
+    const entries = JSON.parse(manifest).entries
+
+    let downloadedDir = {}
+    for (let entry of entries) {
+      const data = await bzz.downloadRawText(entry.hash)
```

Comment:
Using `await` in a `for` loop means the async calls are going to be made in sequence rather than in parallel, to avoid that better use `Promise.all()`, ex:

```js
const entries = Object.values(JSON.parse(manifest).entries)
const downloaded = await Promise.all(entries.map(entry => bzz.downloadRawText(entry.hash)))
const downloadedDir = entries.reduce((acc, entry, i) => {
  acc[entry.path] = {data: downloaded[i]}
  return acc
}, {})
```
This way all the contents are downloaded in parallel and put in an array indexed based on the `entries`, and the `downloadedDir` object can be created by reducing these entries and associating the matching content after all contents have been downloaded.

Reply:
Oh nice. Yeah, that's much better :+1: